### ISSUE TITLE
NPE in isOnVolumeManagedStorage() leads to not being able to extract volumes

### ISSUE THREAD

**Issue by DaanHoogland** [apache/cloudstack#2774]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Storage Motion
~~~

##### CLOUDSTACK VERSION

~~~
4.11.1
~~~

##### CONFIGURATION

In a mixed environment with several advanced zones, extracting a volume failed due to the storage motion strategy org.apache.cloudstack.storage.motion.StorageSystemDataMotionStrategy.canHandle(src,dest) throwing a NullPointerException. root cause is the isVolumeOnManagedStorage() assuming that the dataStore for the destination is Primary while in the case of extractVolume it is an image store. In lab and test environments this seems to happen to go well as both types of storages probably exist with id==1.

##### STEPS TO REPRODUCE

~~~
extract a volume. In the case of a lab environment make sure that no primary storage and image stores exist with the same id. 
~~~

##### EXPECTED RESULTS

~~~
user level: the call returns a url for download
unit level: isVolumeOnMangedStorage should return false or be able to handle the image store case (and still return false?)
~~~

##### ACTUAL RESULTS

~~~
user level: a message 'Failed to copy the volume from the source primary storage pool to secondary storage.' is shown in the UI
unit level: npe
~~~

**Comment by rhtyd** [comment]
Is this fixed @mike-tutkowski ?

**Comment by DaanHoogland** [comment]
this fix has been in since 4.11. I am assuming this was not closed by mistake. we'll open anew if it turns out to be still present.
